daemon/global_statistics.c

```diff
@@ -436,435 +436,953 @@ static void global_statistics_charts(void) {
-                );
+            );
 
-                rd_cached = rrddim_add(st_ram_usage, "cache", NULL, 1, 256, RRD_ALGORITHM_ABSOLUTE);
-                rd_pinned = rrddim_add(st_ram_usage, "collectors", NULL, 1, 256, RRD_ALGORITHM_ABSOLUTE);
-                rd_metadata = rrddim_add(st_ram_usage, "metadata", NULL, 1, 1048576, RRD_ALGORITHM_ABSOLUTE);
-            }
-            else
-                rrdset_next(st_ram_usage);
-
-            API_producers = (collected_number)stats_array[0];
-            pages_on_disk = (collected_number)stats_array[2];
-            populated_pages = (collected_number)stats_array[3];
-            page_cache_descriptors = (collected_number)stats_array[27];
-
-            if (API_producers * 2 > populated_pages) {
-                pinned_pages = API_producers;
-            } else{
-                pinned_pages = API_producers * 2;
-            }
-            cached_pages = populated_pages - pinned_pages;
+            wu->rd_workers_threads_free = rrddim_add(wu->st_workers_threads, "free", NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+            wu->rd_workers_threads_busy = rrddim_add(wu->st_workers_threads, "busy", NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+        }
+        else
+            rrdset_next(wu->st_workers_threads);
 
-            metadata = page_cache_descriptors * sizeof(struct page_cache_descr);
-            metadata += pages_on_disk * sizeof(struct rrdeng_page_descr);
-            /* This is an empirical estimation for Judy array indexing and extent structures */
-            metadata += pages_on_disk * 58;
+        rrddim_set_by_pointer(wu->st_workers_threads, wu->rd_workers_threads_free, (collected_number)(wu->workers_registered - wu->workers_busy));
+        rrddim_set_by_pointer(wu->st_workers_threads, wu->rd_workers_threads_busy, (collected_number)(wu->workers_busy));
+        rrdset_done(wu->st_workers_threads);
+    }
+}
 
-            rrddim_set_by_pointer(st_ram_usage, rd_cached, cached_pages);
-            rrddim_set_by_pointer(st_ram_usage, rd_pinned, pinned_pages);
-            rrddim_set_by_pointer(st_ram_usage, rd_metadata, metadata);
-            rrdset_done(st_ram_usage);
-        }
+static void workers_utilization_reset_statistics(struct worker_utilization *wu) {
+    wu->workers_registered = 0;
+    wu->workers_busy = 0;
+    wu->workers_total_busy_time = 0;
+    wu->workers_total_duration = 0;
+    wu->workers_total_jobs_started = 0;
+    wu->workers_min_busy_time = 100.0;
+    wu->workers_max_busy_time = 0;
+    wu->workers_cpu_enabled = 0;
+
+    size_t i;
+    for(i = 0; i < WORKER_UTILIZATION_MAX_JOB_TYPES ;i++) {
+        wu->per_job_type[i].jobs_started = 0;
+        wu->per_job_type[i].busy_time = 0;
     }
+
+    struct worker_thread *wt;
+    for(wt = wu->threads; wt ; wt = wt->next) {
+        wt->enabled = 0;
+        wt->cpu_enabled = 0;
+    }
+}
+
+static int read_thread_cpu_time_from_proc_stat(pid_t pid, kernel_uint_t *utime, kernel_uint_t *stime) {
```

Comment:
On `FreeBSD` we are receiving this warnings:

```sh
daemon/global_statistics.c:1196:96: warning: unused parameter 'stime' [-Wunused-parameter]
 1196 | static int read_thread_cpu_time_from_proc_stat(pid_t pid, kernel_uint_t *utime, kernel_uint_t *stime) {
      |                                                                                 ~~~~~~~~~~~~~~~^~~~~
```

Please, add :
```sh
UNUSED(stime); 
UNUSED(utime); 
UNUSED(pid);
``` 
after `#else`